src/ipc.cu

```diff
@@ -157,33 +243,35 @@ protected:
         os << "}";
     }
 
-    void read() {
+    void read_schema(const uint8_t *schema_buf, size_t length) {
         if (_fields.size() || _nodes.size()) {
-            throw ParseError("cannot call .read() more than once");
+            throw ParseError("cannot open more than once");
         }
-        read_schema();
-        read_record_batch();
-    }
 
-    void read_schema() {
-        int size = read_msg_size();
-        auto header_buf = read_bytes(size);
-        auto header = parse_msg_header(header_buf);
-        if ( header.body_length > 0) {
-            throw ParseError("schema should not have a body");
-        }
-        parse_schema(header);
+        // Use Arrow to load the schema
+        const auto payload = std::make_shared<arrow::Buffer>(schema_buf, length);
+        auto buffer = std::make_shared<io::BufferReader>(payload);
+        std::shared_ptr<ipc::RecordBatchStreamReader> reader;
+        ipc::RecordBatchStreamReader::Open(buffer, &reader);
```

Comment:
You should check the return value of Open as this is used to indicate any errors.